### ISSUE TITLE
Overflow in csr_matrix.reshape

### ISSUE THREAD

**Issue by akshayka** [scipy/scipy#9697]
My issue is about an overflow error in `csr_matrix.reshape`. The bug appears to be related to the issue addressed in https://github.com/scipy/scipy/pull/9132.

### Reproducing code example:

```
import scipy.sparse as sp


shape = (int(1E5), int(1E5))
m = int(shape[0] / 5)
X = sp.random(shape[0], shape[1], density=1E-6, format='csr')
X.reshape((m, -1))
```

### Error message:

```
Traceback (most recent call last)
<ipython-input-1-65f6128ad0b7> in <module>
      5 m = int(shape[0] / 5)
      6 X = sp.random(shape[0], shape[1], density=1E-6, format='csr')
----> 7 X.reshape((m, -1))

~/virtualenvs/cvxpy-dev/lib/python3.7/site-packages/scipy-1.1.0-py3.7-macosx-10.13-x86_64.egg/scipy/sparse/base.py in reshape(self, *args, **kwargs)
    130                 return self
    131
--> 132         return self.tocoo(copy=copy).reshape(shape, order=order, copy=False)
    133
    134     def resize(self, shape):

~/virtualenvs/cvxpy-dev/lib/python3.7/site-packages/scipy-1.1.0-py3.7-macosx-10.13-x86_64.egg/scipy/sparse/coo.py in reshape(self, *args, **kwargs)
    222
    223         return coo_matrix((new_data, (new_row, new_col)),
--> 224                           shape=shape, copy=False)
    225
    226     reshape.__doc__ = spmatrix.reshape.__doc__

~/virtualenvs/cvxpy-dev/lib/python3.7/site-packages/scipy-1.1.0-py3.7-macosx-10.13-x86_64.egg/scipy/sparse/coo.py in __init__(self, arg1, shape, dtype, copy)
    190             self.data = self.data.astype(dtype, copy=False)
    191
--> 192         self._check()
    193
    194     def reshape(self, *args, **kwargs):

~/virtualenvs/cvxpy-dev/lib/python3.7/site-packages/scipy-1.1.0-py3.7-macosx-10.13-x86_64.egg/scipy/sparse/coo.py in _check(self)
    274                 raise ValueError('column index exceeds matrix dimensions')
    275             if self.row.min() < 0:
--> 276                 raise ValueError('negative row index found')
    277             if self.col.min() < 0:
    278                 raise ValueError('negative column index found')

ValueError: negative row index found
```

### Scipy/Numpy/Python version information:

```
1.1.0 1.15.3 sys.version_info(major=3, minor=7, micro=0, releaselevel='final', serial=0)
```

**Comment by perimosocordiae** [comment]
I'm reasonably sure this has been fixed in a more recent scipy version (by gh-9132), though my attempts to run your example on the current master branch were stymied by gh-9699.

**Comment by akshayka** [comment]
You're right, this was fixed in 1.2.0, so I'm closing the issue.
